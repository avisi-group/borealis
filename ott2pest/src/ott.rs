use pest_derive::Parser;

#[derive(Parser)]
#[grammar = "ott.pest"]
pub(crate) struct Parser;

#[cfg(test)]
mod tests {
    use {
        crate::ott::{Parser, Rule},
        pest::Parser as _,
    };

    #[test]
    fn parse_leroy_jfp96() {
        insta::assert_debug_snapshot!(Parser::parse(
            Rule::main,
            include_str!("../testdata/leroy-jfp96.ott")
        )
        .unwrap());
    }

    #[test]
    fn parse_ocaml() {
        insta::assert_debug_snapshot!(Parser::parse(
            Rule::main,
            include_str!("../testdata/ocaml_syntax.ott")
        )
        .unwrap());
    }

    #[test]
    fn parse_test8() {
        insta::assert_debug_snapshot!(Parser::parse(
            Rule::main,
            include_str!("../testdata/test8.ott")
        )
        .unwrap());
    }

    #[test]
    fn parse_test10() {
        insta::assert_debug_snapshot!(Parser::parse(
            Rule::main,
            include_str!("../testdata/test10.ott")
        )
        .unwrap());
    }

    #[test]
    fn parse_test10st() {
        insta::assert_debug_snapshot!(Parser::parse(
            Rule::main,
            include_str!("../testdata/test10st.ott")
        )
        .unwrap());
    }
}
