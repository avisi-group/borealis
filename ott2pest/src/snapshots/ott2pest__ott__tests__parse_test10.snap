---
source: ott2pest/src/ott.rs
expression: "Parser::parse(Rule::ident_list,\n        include_str!(\"../testdata/test10.ott\")).unwrap()"
---
[
    Pair {
        rule: ident,
        span: Span {
            str: "metavar",
            start: 0,
            end: 7,
        },
        inner: [
            Pair {
                rule: alpha,
                span: Span {
                    str: "m",
                    start: 0,
                    end: 1,
                },
                inner: [],
            },
            Pair {
                rule: alpha,
                span: Span {
                    str: "e",
                    start: 1,
                    end: 2,
                },
                inner: [],
            },
            Pair {
                rule: alpha,
                span: Span {
                    str: "t",
                    start: 2,
                    end: 3,
                },
                inner: [],
            },
            Pair {
                rule: alpha,
                span: Span {
                    str: "a",
                    start: 3,
                    end: 4,
                },
                inner: [],
            },
            Pair {
                rule: alpha,
                span: Span {
                    str: "v",
                    start: 4,
                    end: 5,
                },
                inner: [],
            },
            Pair {
                rule: alpha,
                span: Span {
                    str: "a",
                    start: 5,
                    end: 6,
                },
                inner: [],
            },
            Pair {
                rule: alpha,
                span: Span {
                    str: "r",
                    start: 6,
                    end: 7,
                },
                inner: [],
            },
        ],
    },
    Pair {
        rule: ident,
        span: Span {
            str: "var",
            start: 8,
            end: 11,
        },
        inner: [
            Pair {
                rule: alpha,
                span: Span {
                    str: "v",
                    start: 8,
                    end: 9,
                },
                inner: [],
            },
            Pair {
                rule: alpha,
                span: Span {
                    str: "a",
                    start: 9,
                    end: 10,
                },
                inner: [],
            },
            Pair {
                rule: alpha,
                span: Span {
                    str: "r",
                    start: 10,
                    end: 11,
                },
                inner: [],
            },
        ],
    },
]
